src/zarrzipan/cli.py

```diff
@@ -0,0 +1,335 @@
+    # Define the desired column order and their corresponding keys in the JSON data
+    desired_columns = [
+        ('Dataset Name', 'dataset_name'),
+        ('Pipeline Name', 'pipeline_name'),
+        ('Chunk Shape', 'chunk_shape'),
+        ('Iterations', 'iterations'),
+        ('Compression Ratio', 'size_ratio'),
+        ('Space Saving', 'size_space_saving'),
+        ('Avg Compress Time (ms)', 'time_compress_avg_ms'),
+        ('Avg Decompress Time (ms)', 'time_decompress_avg_ms'),
+        ('Lossiness (MAE)', 'lossiness_mae'),
+    ]
+
+    mean_compression_ratio = np.mean([item['size_ratio'] for item in results])
+
+    # Helper function to format values
+    def format_value(key, value):
+        if key == 'size_ratio':
+            color = 'cyan'
+            if value > mean_compression_ratio + 1:
+                color = 'green'
+            elif value < max(mean_compression_ratio - 1, 1):
+                color = 'red'
+            return Text(f'{value:.2f}x', style=color)
+        if key == 'lossiness_mae':
+            return f'{value:.4f}'
+        if key == 'chunk_shape':
+            return str(value).replace('[', '(').replace(']', ')')
+        if isinstance(value, float):
+            return f'{value:.2f}'
+        return str(value)
+
+    # Create a rich table
+    table = Table(title='Compression Benchmark Results')
+
+    # Add columns to the table
+    for header_name, _ in desired_columns:
+        table.add_column(header_name, justify='right', style='cyan', no_wrap=False)
+
+    # Add rows to the table
+    for data_item in results:
+        row_values = [format_value(key, data_item[key]) for _, key in desired_columns]
+        table.add_row(*row_values)
+
+    return table
+
+
+@click.group()
+@click.option('--verbose', '-v', is_flag=True, help='Print more output.')
+@click.pass_context
+def cli(ctx, verbose):
+    ctx.ensure_object(dict)
+    ctx.obj['v'] = verbose
+
+
+@cli.command()
+@click.pass_context
+@click.option(
+    '--config-file',
+    '-f',
+    default='config.yaml',
+    help='Filepath to config file',
+    type=click.Path(exists=True, dir_okay=False, path_type=Path),
+)
+@click.option(
+    '--data-dir',
+    default=DATA_DIR,
+    help=(f"Directory to cached data in. If not specified '{DATA_DIR}' will be used"),
+    type=click.Path(file_okay=False, path_type=Path),
+)
+@click.option(
+    '--output-file',
+    '-o',
+    required=False,
+    help=(
+        'Filepath to output into. If specified will append to an '
+        'existing file. If not specified a new file will be created '
+        f"within '{OUTPUT_DIR}'"
+    ),
+    type=click.Path(dir_okay=False, path_type=Path),
+)
+@click.option(
+    '--output-dir',
+    default=OUTPUT_DIR,
+    help=(
+        'Directory to put output into. If not specified '
+        f"'{OUTPUT_DIR}' will be used unless output-file is specified"
+    ),
+    type=click.Path(file_okay=False, path_type=Path),
+)
```

Comment:
Why not just write output to stdout? Then the user can redirected it however they wish, including with append if that's what they need.

(I very much have a "never read or write a file directly" bias, and try to ensure everywhere it makes sense that any tooling I create takes input from stdin, outputs to stdout, and uses stderr for any runtime messaging or logs.)

Reply:
I guess it is because you are displaying the results as a table? You could do something like `zarrzipan run | zarrzipan show` if you just wanted to see the table. If you want to write to a file and see the table then `zarrzipan run | tee output.json | zarrzipan show`.